integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/ShowTable/TestShowTable.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.spark.testsuite.ShowTable
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.test.util.QueryTest
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+
+import org.apache.carbondata.common.exceptions.sql.MalformedIndexCommandException
+
+/**
+  * Test class for show tables.
+  */
+class TestShowTable extends QueryTest with BeforeAndAfterAll with BeforeAndAfterEach {
+  val dbName = "testshowtable"
+  override def beforeAll: Unit = {
+    sql(s"drop database if exists $dbName cascade")
+    sql(s"create database $dbName")
+    sql(s"use $dbName")
+  }
+
+  test("test show tables") {
+    sql("create table employee(id string, name string) stored as carbondata")
+    sql("create table employee_part(name string) partitioned by (grade int)")
+    sql("create index employee_si on table employee(name) as 'carbondata'")
+    sql("create materialized view employee_mv as select name from employee group by name")
+    val df = sql("show tables").toDF
```

Comment:
combine rows 41 and 42 and direct do `.collect()` on sql

Reply:
Had done that because show tables returns views as well. Have refactored it now such that we intersection of received rows and expected rows to check.